File: Microservices (H2, JPA, WEB, Lombok, Validation, Devtools) - Building Microservice/accounts/src/main/java/com/didan/microservices/accounts/dto/CustomerDto.java
package com.didan.microservices.accounts.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.Data;

@Data // lombok annotation to generate getters and setters
public class CustomerDto {
	
    @NotEmpty(message = "Name can not be a null or empty")
    @Size(min = 5, max = 30, message = "The length of the customer name should be between 5 and 30")
	private String name;
	
    @NotEmpty(message = "Email address can not be a null or empty")
    @Email(message = "Email address should be a valid value")
	private String email;
	
    @Pattern(regexp = "(^$|[0-9]{10})", message = "Mobile number must be 10 digits")
    @NotEmpty(message = "Mobile number can not be a null or empty")
	private String mobile;
	
	@JsonProperty("accounts")
	private AccountsDto accountsDto;
}

File: GraphQL/src/main/kotlin/com/didan/demo/graphql/validate_input_error_handling/dto/CustomerDto.kt
package com.didan.demo.graphql.validate_input_error_handling.dto

import jakarta.validation.constraints.Min
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Size

data class CustomerDto(
    var id: Int? = null,
    @field:Size(min = 1, message = "name must not be empty")
    var name: String? = null,
    @field:NotNull(message = "age must not be null")
    @field:Min(value = 0, message = "age must be >= 0")
    var age: Int? = null,
    @field:Size(min = 1, message = "city must not be empty")
    var city: String? = null
)
